### ISSUE TITLE
Avoid process execution for network management

### ISSUE THREAD

**Issue by mikalstill** [shakenfist/shakenfist#2066]
It would be nice to avoid executing processes to determine or alter network state, and psutil is close to allowing this but does not support Linux network namespaces. We should keep an eye on https://github.com/giampaolo/psutil/issues/1611 to see if anything lands there.

**Comment by mikalstill** [comment]
https://github.com/larsks/python-netns and https://docs.pyroute2.org/iproute.html both look hopeful? In the case of the latter `NetNS` is an IPRoute API implementation for inside a namespace.

**Comment by mikalstill** [comment]
Closing as part of a bug cleanup sweep of the oldest open issues.

This is now tracked as a full plan: `docs/plans/PLAN-replace-exec-with-netlink.md`. It lands on pyroute2, which is what the 2023-10-19 comment on this issue suggested, and it is gated on `PLAN-network-facade.md` having established a single mutator. Closing in favour of the plan so there is one place tracking the work.
